### ISSUE TITLE
New cop: `Style/RedundantUnless`

### ISSUE THREAD

**Issue by fatkodima** [rubocop/rubocop#11410]
It is a known fact, that `unless` adds cognitive overhead. Personally, I always try to avoid it if `if` looks better/simpler. 
This cop will detect places where `unless` is completely unneeded and can be replaced by `if`.

I was thinking, that there is a similar cop already, but was not able to find it.

For example,
```ruby
# bad
foo unless !bar
foo unless x != y
foo unless x >= 10
foo unless x.positive?

# good
foo if bar
foo if x == y
foo if x < 10
foo if x.negative?
```

I saw at least many offenses in the rubocop itself. Here are only a few examples: https://github.com/rubocop/rubocop/blob/1e5f19a9afddb0e5123c86efb0f5957431c7a6d4/lib/rubocop/cop/layout/trailing_empty_lines.rb#L59
https://github.com/rubocop/rubocop/blob/1e5f19a9afddb0e5123c86efb0f5957431c7a6d4/lib/rubocop/cop/layout/multiline_assignment_layout.rb#L99
https://github.com/rubocop/rubocop/blob/1e5f19a9afddb0e5123c86efb0f5957431c7a6d4/lib/rubocop/cop/metrics/parameter_lists.rb#L82

Related: https://github.com/rubocop/rubocop/issues/11399

There is a kinda related cop already (`Style/NegatedUnless`), but it only checks for simple negations with `!`.

**Comment by vlad-pisanov** [comment]
This is great, although I'm sure about the name. It's not that the `unless` is redundant (as in "duplicated"), it's more that it's confusing / unnatural. `Style/UnnaturalUnless`? `Style/ClearCondition`? idk, can't think of a better name 🤷‍♂️

**Comment by si-lens** [comment]
I like the idea. `unless` is a pain in the ass frequently, especially for non-native English speakers.
However, I agree with @vlad-pisanov  that the cop name could be better.
